internal/pkg/docker/orchestrator/orchestrator.go

```diff
@@ -434,16 +464,19 @@ func (o *Orchestrator) stopTask(ctx context.Context, task Task) error {
 }
 
 // waitForContainerToStart blocks until the container specified by id starts.
-func (o *Orchestrator) waitForContainerToStart(ctx context.Context, id string) error {
+func (o *Orchestrator) waitForContainerToStart(ctx context.Context, id string, isEssential bool) error {
```

Comment:
ah, we take `isEssential` as a boolean parameters, instead of letting the callers swallow the exit error from non-essential container - I assume - because all the callers of `waitForContainerToStart` want the same error handling right? Essentially, we are passing a boolean parameter to avoid code duplication. Is this guess close to what you thought?

This makes sense to me, though I am wary of boolean parameters: among all the cons it has 😂 , it makes code harder to read. For example, it is not obvious what `o.waitForContainerToStart(ctx, opts.ContainerName, true)`  means.

Reply:
yeah your guess is absolutely right 😆   
I want to reduce the duplicacy where each caller has to handle the error accordingly. 
I have two options in this case
1. Adjust the doc comment to make it more clear
2. Let each caller handle the error scenario of non essential container exited.

Which one do you suggest among these two?

Reply:
After a brief search I think there are 3 calls to `waitForContainerToStart`, and only 2 of them need to handle the non-essential case, so the duplication is not bad. Therefore, if you ask me - I would probably have opted for 2. But I will leave the final decision to you 👍🏼